tests/test_utils.py

```diff
@@ -29,6 +31,157 @@ def test_validate_documents(self, git_clone_checkout):
             )
             git_clone_checkout.assert_called_once()
 
+    def test_convert_to_legacy_from_pretraining_messages(
+        self,
+    ):
+        new_dataset: typing.List[MessageSample] = [
+            {
+                "messages": [
+                    {
+                        "role": "system",
+                        "content": "You are a friendly assistant",
+                    },
+                    {
+                        "role": "pretraining",
+                        "content": "<|user|>What is 2+2?<|assistant|>2+2=4",
+                    },
+                ],
+                "group": " test",
+                "dataset": "test-dataset",
+                "metadata": "test",
+            }
+        ]
+        legacy = utils.ensure_legacy_dataset(new_dataset)
+        assert len(legacy) == 1
+        assert legacy[0]["system"] == "You are a friendly assistant"
+        assert legacy[0]["user"] == "What is 2+2?"
+        assert legacy[0]["assistant"] == "2+2=4"
+
+    @pytest.mark.parametrize(
+        "content,exception,match",
+        [
+            ("<|user|>What is 2+2? 2+2=4", ValueError, "<|assistant|>"),
+            ("<|assistant|>2+2=4", ValueError, "<|user|>"),
+            ("<|user|>what is 2+2?<|assistant|>2+2=4", None, ""),
+        ],
+    )
+    def test_invalid_pretraining_messages(
+        self, content: str, exception: Exception | None, match: str
+    ):
+        new_dataset: typing.List[MessageSample] = [
+            {
+                "messages": [
+                    {
+                        "role": "system",
+                        "content": "You are a friendly assistant",
+                    },
+                    {
+                        "role": "pretraining",
+                        "content": content,
+                    },
+                ],
+                "group": " test",
+                "dataset": "test-dataset",
+                "metadata": "test",
+            }
+        ]
+        if exception:
+            with pytest.raises(ValueError, match=match):
```

Comment:
nit: `raises(exception, ...)`?

Reply:
PyTest doesn't like this unfortunately. It requires you to provide the exact exception or else it'll throw some nasty errors.

Reply:
MyPy* not PyTest